src/gui_common/ModalManager.cs

```diff
@@ -125,6 +132,28 @@ public bool HideTopMostPopup()
         return null;
     }
 
+    /// <summary>
+    ///   Attempts to hide the given modal.
+    ///   This won't hide the modal if it's exclusive and doesn't allow closing on escape.
+    /// </summary>
+    /// <param name="popup">Modal to hide</param>
+    /// <returns>True if the modal was hidden</returns>
+    private static bool HideModal(TopLevelContainer popup)
+    {
+        if (popup.Exclusive && !popup.ExclusiveAllowCloseOnEscape)
+            return false;
```

Comment:
I don't think the method should take this into account. It makes sense in the context of `HideTopMostPopup` because that method can also be called by pressing ESC. `HideModal` however is a general method, so its hiding behavior should be absolute.

Reply:
Moved this into the HIdeTopMostPopup, and changed return type to void (as it can only return success now)